src/validation.h

```diff
@@ -744,6 +745,62 @@ bool InvalidateBlock(CValidationState& state, const CChainParams& chainparams, C
 /** Remove invalidity status from a block and its descendants. */
 void ResetBlockFailureFlags(CBlockIndex* pindex) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
+/**
+ * Serialized version of a UTXO set from which an assumed-valid CChainState
+ * can be constructed.
+ */
+class SnapshotMetadata
+{
+public:
+    //! The hash of the block that reflects the tip of the chain for the
+    //! UTXO set contained in this snapshot.
+    uint256 m_base_blockhash;
+
+    //! A hash of the snapshot contents as generated by `GetUTXOStats`.
+    uint256 m_utxo_contents_hash = {};
+
+    //! The number of coins in the UTXO set contained in this snapshot.
+    uint64_t m_coins_count = 0;
+
+    //! Necessary to "fake" the base nChainTx so that we can estimate progress during
```

Comment:
Not clear, you're talking about the progress of background UTXO set here, which is estimated compare to nChainTx of foreground one ?